### ISSUE TITLE
后处理问题

### ISSUE THREAD

**Issue by WuShaogui** [lyuwenyu/RT-DETR#317]
https://github.com/lyuwenyu/RT-DETR/blob/2b88d5d53bcbfbb70329bc9c007fdf7e76cf90dc/rtdetr_pytorch/src/zoo/rtdetr/rtdetr_postprocessor.py#L53

这里使用是模型直接输出结果，未乘上图片尺寸，paddle版本无论是否use_focal_loss，都是使用还原尺寸的，猜测这里应该也是

**Comment by lyuwenyu** [comment]
前面已经乘过了

https://github.com/lyuwenyu/RT-DETR/blob/2b88d5d53bcbfbb70329bc9c007fdf7e76cf90dc/rtdetr_pytorch/src/zoo/rtdetr/rtdetr_postprocessor.py#L38

**Comment by WuShaogui** [comment]
前面乘了后面没用，相当于没乘发自我的手机-------- 原始邮件 --------发件人： Wenyu ***@***.***>日期： 2024年5月23日周四 14:50收件人： lyuwenyu/RT-DETR ***@***.***>抄送： wushaogui ***@***.***>, Author ***@***.***>主    题： Re: [lyuwenyu/RT-DETR] 后处理问题 (Issue #317)
前面已经乘过了
https://github.com/lyuwenyu/RT-DETR/blob/2b88d5d53bcbfbb70329bc9c007fdf7e76cf90dc/rtdetr_pytorch/src/zoo/rtdetr/rtdetr_postprocessor.py#L38

**Comment by YoungsunPan** [comment]
不使用use_focal_loss（用celoss），你有训练成功么。

**Comment by WuShaogui** [comment]
> 不使用use_focal_loss（用celoss），你有训练成功么。

可以收敛，但是同样位置还是有问题，记得把以下

https://github.com/lyuwenyu/RT-DETR/blob/2b88d5d53bcbfbb70329bc9c007fdf7e76cf90dc/rtdetr_pytorch/src/zoo/rtdetr/rtdetr_postprocessor.py#L48

改为

`scores = F.softmax(logits, dim=-1)`

不然最后一个类别无法预测，因为结果被截断了，调试结果如下：

![image](https://github.com/lyuwenyu/RT-DETR/assets/24794718/fd3dfd29-8341-4950-a31a-ec6d93a57a6f)

**Comment by lyuwenyu** [comment]
> > 不使用use_focal_loss（用celoss），你有训练成功么。
> 
> 可以收敛，但是同样位置还是有问题，记得把以下
> 
> https://github.com/lyuwenyu/RT-DETR/blob/2b88d5d53bcbfbb70329bc9c007fdf7e76cf90dc/rtdetr_pytorch/src/zoo/rtdetr/rtdetr_postprocessor.py#L48
> 
> 改为
> 
> `scores = F.softmax(logits, dim=-1)`
> 
> 不然最后一个类别无法预测，因为结果被截断了，调试结果如下：
> 

有兴趣提个PR修一下嘛

**Comment by lyuwenyu** [comment]
这里你训练数据是多少类别？ @WuShaogui

**Comment by WuShaogui** [comment]
> 这里你训练数据是多少类别？ @WuShaogui

训练自己的数据集，就2个类别

**Comment by lyuwenyu** [comment]
你这个https://github.com/lyuwenyu/RT-DETR/pull/320 修改之后有跑精度验证没  精度符合预期嘛

**Comment by WuShaogui** [comment]
> 你这个#320 修改之后有跑精度验证没 精度符合预期嘛

精度符合预期，但是数据集只有50张，类别只有2个，单图目标少于5个，整体比较小，可能没参考意义
